spyder/plugins/onlinehelp/plugin.py

```diff
@@ -7,93 +7,77 @@
-        self.setLayout(layout)
-
-        self.register_widget_shortcuts(self.pydocbrowser.find_widget)
-        self.pydocbrowser.webview.set_zoom_factor(
-            self.get_option('zoom_factor'))
-        self.pydocbrowser.url_combo.setMaxCount(
-            self.get_option('max_history_entries'))
-        self.pydocbrowser.url_combo.addItems(self.load_history())
-
-    #------ Public API ---------------------------------------------------------
-    def load_history(self, obj=None):
-        """Load history from a text file in user home directory"""
-        if osp.isfile(self.LOG_PATH):
-            history = [line.replace('\n', '')
-                       for line in open(self.LOG_PATH, 'r').readlines()]
-        else:
-            history = []
-        return history
+# --- Plugin
+# ----------------------------------------------------------------------------
+class OnlineHelp(SpyderDockablePlugin):
+    """
+    Online Help Plugin.
+    """
 
-    def save_history(self):
-        """Save history to a text file in user home directory"""
-        open(self.LOG_PATH, 'w').write("\n".join( \
-                [to_text_string(self.pydocbrowser.url_combo.itemText(index))
-                 for index in range(self.pydocbrowser.url_combo.count())]))
-
-    #------ SpyderPluginWidget API ---------------------------------------------
-    def toggle_view(self, checked):
-        """Toggle view action."""
-        if checked:
-            if self.pydocbrowser.server is None:
-                self.pydocbrowser.initialize()
-            self.dockwidget.show()
-            self.dockwidget.raise_()
-        else:
-            self.dockwidget.hide()
+    NAME = 'onlinehelp'
+    TABIFY = Plugins.Help
+    CONF_SECTION = NAME
+    CONF_FILE = False
+    WIDGET_CLASS = PydocBrowser
+    LOG_PATH = get_conf_path(NAME)
 
-    def get_plugin_title(self):
-        """Return widget title"""
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
         return _('Online help')
 
-    def get_focus_widget(self):
-        """
-        Return the widget to give focus to when
-        this plugin's dockwidget is raised on top-level
-        """
-        self.pydocbrowser.url_combo.lineEdit().selectAll()
-        return self.pydocbrowser.url_combo
+    def get_description(self):
+        return _(
+            'Browse and search the currently installed modules interactively.')
 
-    def closing_plugin(self, cancelable=False):
-        """Perform actions before parent main window is closed"""
+    def get_icon(self):
+        return self.create_icon('help')
+
+    def on_close(self, cancelable=False):
         self.save_history()
-        self.set_option('zoom_factor',
-                        self.pydocbrowser.webview.get_zoom_factor())
-        self.pydocbrowser.quit_server()
+
+        # FIXME: move the zoom trigger to an option in the widget?

```

Comment:
Please address this FIXME

Reply:
Deleting comment. Not fixable since changing that will not emit a signal.